### PR TITLE
Port the logging categories to categorised logging

### PR DESCRIPTION
**Pull request by sowelipililimute**
This makes it easier to get at Prism's logs specifically (I didn't break the logging categories into anything more granular than that.)

Also includes installation of metadata for compatibility with kdebugsettings and similar programs on *nixes that aren't macOS.

### CODE REVIEW

**Comment by Scrumplex** [comment]
This needs a sign-off from your end, to apply with our DCO policy.

```
git rebase HEAD~1 --signoff
git push --force
```

**Comment by sowelipililimute** [comment]
done

File: launcher/CMakeLists.txt [hunk 1]
```diff
@@ -568,6 +568,23 @@ set(LOGIC_SOURCES
     ${ATLAUNCHER_SOURCES}
 )
 
+ecm_qt_declare_logging_category(LOGIC_SOURCES
+    HEADER prismlauncherdebug.h
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt]: I would prefer if this was just called `launcherlog.h`. We have some downstream forks with different branding and don't want to make their lives a living hell

File: launcher/CMakeLists.txt [hunk 2]
```diff
@@ -568,6 +568,23 @@ set(LOGIC_SOURCES
...
+    IDENTIFIER PRISM_LAUNCHER
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt]: This identifier should be defined as a variable in `program_info/CMakeLists.txt` and referenced here

File: launcher/CMakeLists.txt [hunk 3]
```diff
@@ -568,6 +568,23 @@ set(LOGIC_SOURCES
...
+    CATEGORY_NAME prismlauncher
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt]: If this doesn't have to be lowercase, use `${Launcher_Name}`

File: launcher/CMakeLists.txt [hunk 4]
```diff
@@ -568,6 +568,23 @@ set(LOGIC_SOURCES
...
+    DEFAULT_SEVERITY Info
+    EXPORT PrismLauncher
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt]: `${Launcher_Name}`

File: launcher/CMakeLists.txt [hunk 5]
```diff
@@ -568,6 +568,23 @@ set(LOGIC_SOURCES
...
+    DESCRIPTION "The Prism Launcher for Minecraft"
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt]: `${Launcher_DisplayName}` instead of `Prism Launcher`

**Review by Scrumplex — COMMENTED** [review]
Preliminary review

**Comment by sowelipililimute** [comment]
done

**Comment by Scrumplex** [comment]
CC @flowln you were working on this as well. Not sure why you closed it back then, though. Maybe you can still take a look when you find the time :eyes:

File: CMakeLists.txt [hunk 6]
```diff
@@ -64,6 +64,11 @@ find_package(ECM REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH "${ECM_MODULE_PATH};${CMAKE_MODULE_PATH}")
 include(CTest)
 include(ECMAddTests)
+include(ECMQtDeclareLoggingCategory)
+if(UNIX AND NOT APPLE)
+    # only used for kdebugsettings compatibility for logging categories
+    include(KDEInstallDirs)
```
Comment (flowln) [inline CMakeLists.txt:70]: this doesn't seem to work with Qt6 yet (maybe when KF6 is out, for now there's only [an alpha version that we shouldn't use](https://api.kde.org/ecm/kde-module/KDEInstallDirs6.html)

**Review by flowln — COMMENTED** [review]
i won't say you _need_ to change it, since it seems like a lot of work, but the includes are mixed in the out-of-tree includes (`<...>`), and it looks ugly.

> CC @flowln you were working on this as well. Not sure why you closed it back then, though.

That's because it was the second time I tried to do something to improve CLI, and the second time i was backstabbed by another PR created later than mine and fucking things up for me. I still think having the ability to quiet out certain components from the logs would be helpful for development, but would anyone else agree with me?

**Comment by Scrumplex** [comment]
> second time i was backstabbed by another PR created later than mine and fucking things up for me

sorry >.<

**Comment by Scrumplex** [comment]
This needs a rebase on/merge from develop

**Comment by flowln** [comment]
I'll close this for now, seeing the lack of activity. If you want to work on it again, feel free to re-open the PR!

And thank you for putting work into it, it's very appreciated :)

**Pull request closed without merging**